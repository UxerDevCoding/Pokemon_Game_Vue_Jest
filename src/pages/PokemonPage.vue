<template>
    <h1 v-if="!pokemon"> Cargando ...</h1>
    <div v-else>
        <h1>¿Qué Pokémon es este?</h1>
        <pokemon-image 
            :pokemonId="pokemon.id" 
            :showPokemon="showPokemon"/>

        <pokemon-type 
            :pokemons="pokemonArr"
            @selection="checkAnswer" />
    </div>

    <div v-if="showAnswer">
        <h2>{{message}}</h2>
        <button @click="newGame">
            Nuevo Juego
        </button>
   </div>
</template>

<script>


import PokemonImage from '@/components/PokemonImage.vue'
import PokemonType from '@/components/PokemonType.vue'

import getPokemonType from '@/helpers/getPokemonType'

export default {
    components: {
        PokemonImage,
        PokemonType,
    },
    data() {
        return {
            pokemonArr: [],
            pokemon: null,
            showPokemon: false,
            showAnswer: false,
            message: '',

        }
    },
    methods: {
        async mixPokemonArr() {
            this.pokemonArr = await getPokemonType()

            const rndInt = Math.floor(Math.random() * 4)
            this.pokemon = this.pokemonArr[rndInt]
        },
        checkAnswer(pokemonId) {
             this.showPokemon = true
            this.showAnswer = true;

            if (pokemonId === this.pokemon.id) {
                this.message = `Pokémon correcto, es "${this.pokemon.name}"`
            } else {
                this.message = `Pokémon incorrecto, era "${this.pokemon.name}"`  
            }

        },
    newGame() {

        this.showPokemon = false
        this.showAnswer = false
        this.pokemonArr = []
        this.pokemon = null
        this.mixPokemonArr()     
        }
    },
    mounted() {
        this.mixPokemonArr()

    }

}
</script>

<style scoped>

button {
    background-color: #77e9c3;
    border: none;
    padding: 6px 8px;
    font-size: 14px;
    font-weight: bold;
    cursor: pointer;
}

</style>