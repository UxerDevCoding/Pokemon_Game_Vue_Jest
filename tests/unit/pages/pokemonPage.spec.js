import { mount, shallowMount} from '@vue/test-utils';
import PokemonPage from '@/pages/PokemonPage';
import {pokemons} from '../components/mocks/pokemonsMock'


describe('PokemonPage Component', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallowMount(PokemonPage);
    })
    

    test('should match with the snapshot', () => {

        expect(wrapper.html()).toMatchSnapshot()

    });

    test('should call mixPokemonArray in Mounted', () => {
        
        const mixPokemonArrSpy = jest.spyOn(PokemonPage.methods, 'mixPokemonArr');
        shallowMount(PokemonPage)
        expect(mixPokemonArrSpy).toHaveBeenCalled();

    })
    
  
    test('checkanswer should return the correct answer ', async () => {
      
        const wrapper = shallowMount( PokemonPage, {
            data() {
                return {
                    pokemonArr: pokemons,
                    pokemon: pokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        })

    await wrapper.vm.checkAnswer(5)

    expect( wrapper.find('h2').exists() ).toBeTruthy()
    expect( wrapper.vm.showPokemon ).toBe(true)
    expect( wrapper.find('h2').text() ).toBe(`Pok√©mon incorrecto, era "${ pokemons[0].name }"`)

   
})

    


})