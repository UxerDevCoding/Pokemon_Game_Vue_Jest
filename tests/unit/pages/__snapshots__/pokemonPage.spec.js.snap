// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage Component should match to snapchot when pokemons are loaded 1`] = `
<h1> Cargando ...</h1>
<!--v-if-->
`;

exports[`PokemonPage Component should match with the snapshot 1`] = `
<h1> Cargando ...</h1>
<!--v-if-->
`;
